DataSetExplorer/DataSetBuilder/Model/DataSetInstance.cs

```diff
@@ -23,14 +23,24 @@ internal DataSetInstance(string codeSnippetId, string link, string projectLink,
             Type = type;
             
             Annotations = new HashSet<DataSetAnnotation>();
-            MetricFeatures = metricFeatures;
+            SetMetricFeatures(metricFeatures);
             Validate();
         }
 
         private DataSetInstance()
         {
         }
 
+        private void SetMetricFeatures(Dictionary<CaDETMetric, double> metricFeatures)
+        {
+            if (metricFeatures == null) return;
```

Comment:
You can integrate this logic into the setter (https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-properties).